Tests: USBSerial: Handle minimal printf limitations

fkjagodzinski opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

In a line coding test, host changes the line coding based on data received from DUT via the `USBSerial::printf()`. A fixed-size payload, equal to `LINE_CODING_STRLEN`, is required by the host side. When the minimal printf is used, fixed-width messages can not be easily generated on the fly. Zero-fill or width specifiers are not supported by the minimal printf.

Update the host side to stop reading on a defined payload delimiter and allow variable-width messages.

For more details, please see a related issue #12718.

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->
None

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

@jamesbeyond, @evedon, @maciejbocianski 

----------------------------------------------------------------------------------------------------------------

evedon reviewed (dismissed):
I think we should have the same test for minimal-printf and standard C printf and change the test assertion `TEST_ASSERT_EQUAL_INT(LINE_CODING_STRLEN, rc);`
You could test that rc is between a minimal value and LINE_CODING_STRLEN.

fkjagodzinski committed:
Tests: USBSerial: Handle minimal printf limitations

In a line coding test, host changes the line coding based on data received
from DUT via the USBSerial::printf(). A fixed-size payload, equal to
LINE_CODING_STRLEN, is required by the host side. When the minimal printf is
used, fixed-width messages can not be easily generated on the fly. Zero-fill
or width specifiers are not supported by the minimal printf.

Update the host side to stop reading on a defined payload delimiter and
allow variable-width messages.

evedon commented on TESTS/usb_device/serial/main.cpp:
```diff
@@ -84,7 +84,8 @@
 #define USB_RECONNECT_DELAY_MS 1
 
-#define LINE_CODING_STRLEN 13 // 6 + 2 + 1 + 1 + 3 * comma
+#define LINE_CODING_SEP (',')
```
This is not needed in this file

evedon reviewed (approved):
Minor comment but code change looks good.
Thanks!

rajkan01 reviewed (approved):
LGTM
